haystack/components/agents/agent.py

```diff
@@ -357,7 +367,11 @@ async def run_async(
                 tool_invoker_result = await AsyncPipeline._run_component_async(
                     component_name="tool_invoker",
                     component={"instance": self._tool_invoker},
-                    component_inputs={"messages": llm_messages, "state": state},
+                    component_inputs={
+                        "messages": llm_messages,
+                        "state": state,
+                        "streaming_callback": streaming_callback,
```

Comment:
@Amnah199 huh I just realized that in this scenario we'd be passing an async streaming callback to the sync version of the ToolInvoker run method. This is because ToolInvoker doesn't have a `run_async` method. 

In this scenario we may need to leave out the `streaming_callback` from there for now and work on implementing a `run_async` method for the ToolInvoker. 

cc @julian-risch we may need to bump the prio of this issue https://github.com/deepset-ai/haystack/issues/9008